libraries/apollo-gradle-plugin-external/src/main/kotlin/com/apollographql/apollo3/gradle/internal/DefaultApolloExtension.kt

```diff
@@ -184,6 +184,16 @@ abstract class DefaultApolloExtension(
       }
 
       maybeLinkSqlite()
+
+      checkForLegacyJsTarget()
+    }
+  }
+
+  private fun checkForLegacyJsTarget() {
+    val kotlin = project.extensions.findByName("kotlin") as? KotlinMultiplatformExtension
+    val hasLegacyJsTarget = kotlin?.targets?.any { target -> target is KotlinJsTarget && target.irTarget == null } == true
+    check(!hasLegacyJsTarget) {
+      "Apollo: LEGACY js target is not supported by Apollo, please use IR."
```

Comment:
We need to add something in the CHANGELOG about this, certainly link to the [Kotlin changelog](https://kotlinlang.org/docs/whatsnew-eap.html#kotlin-js) too